curriculum/test/test-challenges.js

```diff
@@ -86,59 +88,141 @@ spinner.text = 'Populate tests.';
   });
 
-  const testLangs = testedLangs();
+async function setup() {
+  if (process.env.npm_config_superblock && process.env.npm_config_block) {
+    throw new Error(`Please do not use both a block and superblock as input.`);
+  }
 
-  const challenges = await Promise.all(
+  // liveServer starts synchronously
+  liveServer.start({
+    host: '127.0.0.1',
+    port: '8080',
+    root: path.resolve(__dirname, 'stubs'),
+    mount: [['/js', path.join(clientPath, 'static/js')]],
+    open: false,
+    logLevel: 0
+  });
+  browser = await puppeteer.launch({
+    args: [
+      // Required for Docker version of Puppeteer
+      '--no-sandbox',
+      '--disable-setuid-sandbox',
+      // This will write shared memory files into /tmp instead of /dev/shm,
+      // because Docker’s default for /dev/shm is 64MB
+      '--disable-dev-shm-usage'
+      // dumpio: true
+    ]
+  });
+  global.Worker = createPseudoWorker(await newPageContext(browser));
+  page = await newPageContext(browser);
+  await page.setViewport({ width: 300, height: 150 });
+  const testLangs = testedLangs();
+  const challengesForLang = await Promise.all(
     testLangs.map(lang => getChallenges(lang))
   );
 
-  describe('Check challenges', function() {
-    before(async function() {
-      spinner.text = 'Testing';
-      this.timeout(50000);
-      liveServer.start({
-        host: '127.0.0.1',
-        port: '8080',
-        root: path.resolve(__dirname, 'stubs'),
-        mount: [['/js', path.join(clientPath, 'static/js')]],
-        open: false,
-        logLevel: 0
-      });
-      browser = await puppeteer.launch({
-        args: [
-          // Required for Docker version of Puppeteer
-          '--no-sandbox',
-          '--disable-setuid-sandbox',
-          // This will write shared memory files into /tmp instead of /dev/shm,
-          // because Docker’s default for /dev/shm is 64MB
-          '--disable-dev-shm-usage'
-          // dumpio: true
-        ]
-      });
-      global.Worker = createPseudoWorker(await newPageContext(browser));
-      page = await newPageContext(browser);
-      await page.setViewport({ width: 300, height: 150 });
-    });
-    after(async function() {
-      this.timeout(30000);
-      if (browser) {
-        await browser.close();
+  // the next few statements create a list of all blocks and superblocks
+  // as they appear in the list of challenges
+  const filters = {
+    blocks: [],
+    superBlocks: []
+  };
+
+  challengesForLang[0].challenges.forEach(chal => {
+    if (filters.blocks.indexOf(chal.block) < 0) {
+      filters.blocks.push(chal.block);
+    }
+
+    if (filters.superBlocks.indexOf(chal.superBlock) < 0) {
+      filters.superBlocks.push(chal.superBlock);
+    }
+  });
+
+  // the next few statements will filter challenges based on command variables
+  if (process.env.npm_config_superblock) {
+    const filter = stringSimilarity.findBestMatch(
+      process.env.npm_config_superblock,
+      filters.superBlocks
```

Comment:
```suggestion
      targetSuperBlockStrings
```
`filter` was a little overloaded.